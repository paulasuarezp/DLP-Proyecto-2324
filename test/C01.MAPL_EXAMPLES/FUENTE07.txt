// Dado un nยบ, imprime 1 si es primo y 0 si no lo es

int numero;
int divisor;

read numero;

if (numero < 2)
    print 0;
else {  // Se puede halt en vez de else
    divisor = 2;
    while (divisor < numero) {
        if (numero % divisor == 0) {
            print 0;
            halt;
        }
        divisor = divisor + 1;
    }
    print 1;
}
