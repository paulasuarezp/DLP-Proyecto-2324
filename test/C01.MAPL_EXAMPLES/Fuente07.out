' Compilador mini Eiffel.
' Aquilino A. Juan.
' Versi√≥n 2024.2 (c)
#source "Fuente07.me"
#global numero: int
#global divisor: int
#func main
#ret void
CALL main
HALT

#line 15
main:

#line 19
PUSHA 0
INI
STOREI

#line 20
PUSHA 0
LOADI
PUSHI 2
LTI
JZ else_0000

#line 21
PUSHI 0
OUTI
JMP endif_0000
else_0000:

#line 24
' inicio from_0001

#line 24
PUSHA 2
PUSHI 2
STOREI

#line 25
UntilCond_0001:
PUSHA 2
LOADI
PUSHA 0
LOADI
GEI
JNZ UntilEnd_0001

#line 26
PUSHA 0
LOADI
PUSHA 2
LOADI
modI
PUSHI 0
EQI
JZ else_0002

#line 27
PUSHI 0
OUTI

#line 28
RET 0, 0, 0
JMP endif_0002
else_0002:
endif_0002:

#line 30
PUSHA 2
PUSHA 2
LOADI
PUSHI 1
ADDI
STOREI
JMP UntilCond_0001
UntilEnd_0001:

#line 32
PUSHI 1
OUTI
endif_0000:

#line 32
RET 0, 0, 0
